javascript/javascript2/week1/spiritAnimal.js

```diff
@@ -0,0 +1,20 @@
+const spiritAnimal=["The crying butterfly", "The fullmoon wolf", "The powerful Tiger", "The angry bird"];//given some random names
+const inputName=document.getElementById("name");//inputName what we are typing
+const input=document.getElementById("input");//to the output using div id
+const randomName=function()//function calling
+{
+
+    if(inputName==="")
+    {
+        input.innerHTML="Please enter name here";
+    }
+    else
+    {
+        const randomNumber = Math.floor(Math.random()*spiritAnimal.length);//logic for random selection
+        const randomSpiritAnimalName = spiritAnimal[randomNumber];//logic here for random selection
+        input.innerHTML=`${inputName}` + `${randomSpiritAnimalName}`;
```

Comment:
This format string is too complex, your mixing up the usefulness of the backtick operator ` with the old way of concatenating strings

The 2 ways your mixing up are, using backticks 
```
`${inputName} ${randomSpiritAnimalName}`
```

and concatenating strings
```
inputName + " " + randomSpiritAnimalName
```

The usefulness of the backticks is that you can do the string formatting inline in the string instead of needing to resort to concatenating strings, it's also *much* more readable.  You should mostly use it when you can